### PR TITLE
Move crashes from centerline to location centroid when associating them to a service road polygon

### PR DESCRIPTION
**Pull request by frankhereford**
Work to close cityofaustin/atd-data-tech/issues/6915.

As discussed in the grooming session, completing this work will be enabled by a potential upgrade to Hasura 2.0. Additionally, there was discussion of working around not being able to track VOLITILE functions by submitting SQL via Hasura 1.0.

### CODE REVIEW

**Comment by frankhereford** [comment]
I have added the new view this needs to the staging hasura /pg deployment.

I have tested this by running the app.py on my local machine and it's doing the right thing to the crash in my own hasura instance. I will continue testing on the staging deployment.

**Review by patrickm02L — APPROVED** [review]
Appreciate the thorough notes. SHIP IT!

File: atd-events/crash_update_location/app.py [hunk 1]
```diff
@@ -327,18 +406,23 @@ def hasura_request(record: str) -> bool:
...
     old_location_id = get_location_id(data)
+    relocate_crash_to_service_road_centroid = False
 
     # Check if this crash is a main-lane
     if is_crash_mainlane(crash_id):
         if(nonproper_level_5_directional_polygon := is_crash_nonproper_and_directional(crash_id)):
             new_location_id = nonproper_level_5_directional_polygon
+            relocate_crash_to_service_road_centroid = True
+            centroid = get_centroid_for_location(new_location_id)
+            set_crash_position(crash_id, centroid)
```
Comment (mateoclarke) [inline atd-events/crash_update_location/app.py:417]: 👍

**Review by mateoclarke — APPROVED** [review]
🚢 💨 👏 

fortune-algorithm

**Pull request merged**